maven-scm-providers/maven-scm-providers-git/maven-scm-provider-git-commons/src/main/java/org/apache/maven/scm/provider/git/repository/GitScmProviderRepository.java

```diff
@@ -246,21 +259,11 @@ private String getUrl(RepositoryUrl repoUrl) {
             }
 
             if (userName != null && userName.length() > 0) {
-                String userInfo = userName;
+                String userInfo = GitUtil.encodeURIComponent(userName);
                 if (password != null && password.length() > 0) {
-                    userInfo += ":" + password;
+                    userInfo += ":" + GitUtil.encodeURIComponent(password);
```

Comment:
This is a separate improvement, right? If so, can you separate it out? I'd merge it separately.

Reply:
It is not an improvement - it is key to get the password URL encoded. The only change is that it is using the central util method for encoding now. The inconsistent encoding before was the root cause of the vulnerability issue.